src/Utilities/product_of_sets.jl

```diff
@@ -34,28 +35,60 @@ end
 
 Product of scalar sets in the order the constraints are added, mixing the
 constraints of different types.
+
+Use [`@mix_of_scalar_sets`](@ref) to generate a new subtype.
+
+If the sets are all scalar, use this type instead of [`OrderedProductOfSets`](@ref)
```

Comment:
I wouldn't say it's more efficient. In fact, for Clp I ended up using `OrderedProductOfSets` as it was faster.